### DISCUSSION TITLE
[cloud_firestore] Add custom configuration option of `experimentalForceLongPolling`

### DISCUSSION THREAD

**Discussion by alexagat** [firebase/flutterfire#6634] [Feature request]
After deploying a Flutter web application many users report not being able to use it due to failing requests to read/write data. Specifically the gRPC network requests to `https://firestore.googleapis.com/google.firestore.v1.Firestore/Listen/channel?...` are the ones failing.

All of the users experiencing this issues are on a corporate network or have some Firewall layer enabled. This same issue occurs with non-Flutter web applications built in Angular/React using the `firebase-js-sdk` SDK, but the javascript sdk allows for setting an option of `experimentalForceLongPolling` which seems to help. 

Documentation on JS SDK FirebaseFirestore options: https://firebase.google.com/docs/reference/js/firebase.firestore.Settings#experimentalforcelongpolling

Related issue against the JS SDK: https://github.com/firebase/firebase-js-sdk/issues/1674 https://github.com/firebase/firebase-js-sdk/issues/2526

Unfortunately, `experimentalForceLongPolling` does not appear to be an option for the FlutterFire SDK. Can it be added?

**Comment by TechnoMarc3** [comment]
Really nice, that after two years they didnt add this IMPORTANT feature, good job!

**Comment by RizanPSTU** [comment]
Any way to add this?
